recipe/meta.yaml

```diff
@@ -1,33 +1,32 @@
 {% set name = "ruamel.yaml" %}
-{% set version = "0.17.21" %}
+{% set version = "0.18.6" %}
```

Comment:
I wonder if this version doesn't break some of the [downstream packages](https://metayaml-conda.fly.dev/metayaml/reverse_dependencies?name=ruamel.yaml) because of this warning https://inspector.pypi.io/project/ruamel-yaml/0.18.6/packages/29/81/4dfc17eb6ebb1aac314a3eb863c1325b907863a1b8b1382cdffcb6ac0ed9/ruamel.yaml-0.18.6.tar.gz/ruamel.yaml-0.18.6/PKG-INFO#line.65

Reply:
Good point

Reply:
`0.18` was released in October 2023. The packages we have that depend on ruamel either are at versions released in 2024 or do not have new versions. This seems relatively safe to me.

Reply:
Hmm not sure I'm following this logic... they could still be using the deprecated functions in both those cases no? It could be good to at least test the conda tools with the update

Reply:
'conda' and 'constructor' have upper bounds, and this package request comes from 'conda'. 
Also, all these packages are in conda-forge, which has had ruamel 0.18 since October 2023.

Reply:
yes but the upper bound is <0.19 so wouldn't restrict this version. Well, if it's on conda-forge already we would have heard about it.

Reply:
I added a commit to run downstreams, for sanity.

Reply:
Thank you, Charles

Reply:
`conda-auth 0.2.1 has requirement ruamel-yaml<0.18, but you have ruamel-yaml 0.18.6.`
Our recipe doesn't have an upper bound https://github.com/AnacondaRecipes/conda-auth-feedstock/blob/main/recipe/meta.yaml#L28 but it must have <18, see https://github.com/conda-incubator/conda-auth/blob/main/recipe/meta.yaml#L25

Reply:
All other downstream packages are fine but `oauthenticator 15.1.0` doesn't have **py312** variant:
```
WARNING: Skipping downstream test for spec oauthenticator 15.1.0; was unsatisfiable.  Error was Unsatisfiable dependencies for platform osx-arm64: {MatchSpec("python[version='>=3.12,<3.13.0a0']"), MatchSpec("ruamel.yaml==0.18.6=py312h8a0c2ca_0")}
Encountered problems while solving:
  - package ruamel.yaml-0.18.6-py312h8a0c2ca_0 requires python >=3.12,<3.13.0a0, but none of the providers can be installed
```
Not critical but we should update or rebuild it later

Reply:
I opened a ticket for oauthenticator. conda-auth is only about pip check failing. The code itself is compatible with ruamel 0.18.